usr.sbin/uhsoctl/uhsoctl.c

```diff
@@ -360,19 +360,33 @@ logger(int pri, const char *fmt, ...)
 static int
 ifaddr_ad(unsigned long d, const char *ifnam, struct sockaddr *sa, struct sockaddr *mask)
 {
-	struct ifaliasreq req;
+	struct ifreq ifr;
+	struct ifaliasreq ifra;
+	caddr_t req;
 	int fd, error;
 
 	fd = socket(AF_INET, SOCK_DGRAM, 0);
 	if (fd < 0)
 		return (-1);
 
-	memset(&req, 0, sizeof(struct ifaliasreq));
-	strlcpy(req.ifra_name, ifnam, sizeof(req.ifra_name));
-	memcpy(&req.ifra_addr, sa, sa->sa_len);
-	memcpy(&req.ifra_mask, mask, mask->sa_len);
+	switch (d) {
+	case SIOCAIFADDR:
+		memset(&ifra, 0, sizeof(ifra));
+		strlcpy(ifra.ifra_name, ifnam, sizeof(ifra.ifra_name));
+		memcpy(&ifra.ifra_addr, sa, sa->sa_len);
+		memcpy(&ifra.ifra_mask, mask, mask->sa_len);
```

Comment:
I find myself a little tempted to start with a designated initializer and then strlcpy the name in, but this is probably more standard.

Reply:
We have to do `memset()` in the case of `SIOCDIFADDR` below (there are no other fields initialised than `ifr_name`), so I'd be tempted to use `memset()` in both cases.